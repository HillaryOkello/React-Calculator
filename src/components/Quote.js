import React from 'react';
import Navbar from './Navbar';

const style = {
  padding: '5rem',
};

const Quote = () => (
  <>
    <Navbar />
    <h3 style={style}>
      Mathematics is not about numbers, equations, computations,
      or algorithms: it is about understanding. â€“William Paul Thurston
    </h3>
  </>
);

export default Quote;
