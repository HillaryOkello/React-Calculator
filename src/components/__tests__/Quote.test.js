import React from 'react';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Quote from '../Quote';
import '@testing-library/jest-dom/extend-expect';

const history = createBrowserHistory();

it('renders correctly', () => {
  const tree = renderer.create(
    <Router history={history}>
      <Quote />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains NavBar', () => {
  const { getByText } = render(
    <Router history={history}>
      <Quote />
    </Router>,
  );
  expect(getByText('Math Magicians')).toBeInTheDocument();
});
