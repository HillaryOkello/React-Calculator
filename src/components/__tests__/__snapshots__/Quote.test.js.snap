// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="top"
  >
    <h1>
      Math Magicians
    </h1>
    <div
      className="links"
    >
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="link"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        className="link"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </div>,
  <h3
    style={
      Object {
        "padding": "5rem",
      }
    }
  >
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. â€“William Paul Thurston
  </h3>,
]
`;
